File: backend/src/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProjectBase(BaseModel):
    name: str = Field(
        min_length=1,
        max_length=100,
    )

    description: str | None = None

class ProjectCreate(ProjectBase):
    pass

class ProjectUpdate(BaseModel):
    name: str | None = Field(
        default=None,
        min_length=1,
        max_length=100,
    )

    description: str | None = None

class ProjectResponse(ProjectBase):
    id: int
    owner_id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/src/types/project.ts
/** Mirrors backend schemas/project.py — ProjectCreate */
export interface ProjectCreate {
  name: string;
  description?: string | null;
}

/** Mirrors backend schemas/project.py — ProjectUpdate */
export interface ProjectUpdate {
  name?: string | null;
  description?: string | null;
}

/** Mirrors backend schemas/project.py — ProjectResponse */
export interface ProjectResponse {
  id: number;
  name: string;
  description: string | null;
  owner_id: number;
  created_at: string;
  updated_at: string;
}
